tests/test_references.py

```diff
@@ -121,18 +130,27 @@ def test_missing_reference_with_markdown_text():
 def test_missing_reference_with_markdown_id():
     """Check unmapped explicit references with Markdown in the identifier."""
     run_references_test(
-        url_map={"NotFoo": "foo.html#NotFoo"},
-        source="[Foo][*oh*]",
-        output="<p>[Foo][*oh*]</p>",
-        unmapped=["*oh*"],
+        url_map={"Foo": "foo.html#Foo", "NotFoo": "foo.html#NotFoo"},
+        source="[Foo][*NotFoo*]",
+        output="<p>[Foo][*NotFoo*]</p>",
+        unmapped=["*NotFoo*"],
     )
 
 
 def test_missing_reference_with_markdown_implicit():
     """Check that implicit references are not fixed when the identifier is not the exact one."""
     run_references_test(
-        url_map={"Foo": "foo.html#Foo"},
-        source="[`Foo`][]",
-        output="<p>[<code>Foo</code>][]</p>",
-        unmapped=[],
+        url_map={"Foo-bar": "foo.html#Foo-bar"},
+        source="[*Foo-bar*][] and [`Foo`-bar][]",
+        output="<p>[<em>Foo-bar</em>][*Foo-bar*] and [<code>Foo</code>-bar][]</p>",
```

Comment:
This first part doesn't look great, but it was already like that

Reply:
You mean `[*Foo-bar*][]` was already transformed as `[<em>Foo-bar</em>][*Foo-bar*]` by the current code?

Reply:
Yes

Reply:
As agreed, we'll deal with this in another issue/PR.